ui: cluster-ui integration

koorosh opened the pull request:
Note: Most of the commits are transferred "as is" from "cockroachdb/ui" repository and only two
last commits (be2cf6f and 7391346) include changes related to actual integration of package and adjusting
build tools.

Initially db-console referenced to `cluster-ui` package as an external package that
is hosted in another repository and required to update version of this package
every time new changes are made.

Now, when `cluster-ui` package is extracted from `cockroachdb/ui` repository and
moved in this repo (under `pkg/ui/cluster-ui`), it is possible to reference to
this package as a linked dependency and immediately observe all changes.

To do this following changes were made:
- db-console references to cluster-ui as a linked dependency and doesn't rely
on specific version
- `cluster-ui` package relies on local version of crdb-protobufjs-client package
instead of specific version
- Make commands related to db-console are extended to take into account nested
`cluster-ui` package. It runs tests, linting and watch commands for both packages.

One notable workaround has to mentioned... yarn installation for db-console
fails during first run because it cannot properly consume symlinked `cluster-ui`
dependency and only with second attempt it passes without issues (this is known
issue in yarn project).

Depends on: https://github.com/cockroachdb/yarn-vendored/pull/73

koorosh committed:
db-console: optimize highlight.js imports

By default highlight.js package imports lots of modules
with support of different syntax.
To load only SQL syntax, highlight.js core module imported
and then sql syntax module is registered.

koorosh committed:
db-console: optimize webpack builds with production mode

Initially, webpack config was defined with hard coded "development"
mode that isn't sufficient for production.
Now, webpack's "mode" option is provided as an argument in package.json
to set production mode for builds.
Production mode provides a set of optimizations (minification, etc).

koorosh committed:
db-console: optimize antd imports

antd library by default imports lib modules, bundled files,
and lots of styles that dramatically increase bundle size.
To fix this, babel-import plugin is used, it allows automatically
import lib modules instead of entire package and also imports
all relevant styles for imported component (see:
https://github.com/ant-design/babel-plugin-import#-libraryname-antd-style-true-)
Along with this, all Antd modules should be imported as import { xxx } from "antd"

koorosh committed:
db-console: optimize moment.js imports

By default moment.js package imports all available locales
which increase bundle size.
Currently, only US locale is used and `moment-locales-webpack-plugin`
is used to exclude all except US (default locale) from
build.

dhartunian committed:
Admin UI components rename (#154)

This PR changes the name of the "admin-ui-components"
project to "cluster-ui".

* renamed `admin-ui-components` directory
* renamed usages of `admin-ui-components`
* updated docs to reflect cluster ui name

Code changes were made where appropriate to reflect the
change.

Omitted from this change are:
* any CSS modifications
* any changes to Docs links and URLs since those are
managed separately

dhartunian committed:
v0.1.38

koorosh committed:
v0.1.39

koorosh committed:
v0.1.40

koorosh committed:
cluster-ui: StatementDetailsConnected component

StatementDetailsConnected component is a bridge between
redux store and StatementDetails component to provide
required data and handle callbacks from StatementDetails.

This change does replicate the same behavior as before,
except the change how analytics actions dispatched.
Now these actions dispatched and handled by redux saga
as a side effect (previously analytics service was directly
called from StatementDetails component).

koorosh committed:
cluster-ui: nodes redux functionality

These change extends redux functionality with
two more reducers: nodes and node liveness statuses.
Reducers and sagas for these slices are implemented
in the same way as statements redux functionality,
with only one exception:
- nodes saga performs transformation for received
data to replicate the same logic as were before.

koorosh committed:
cluster-ui: nodes redux functionality

These change extends redux functionality with
two more reducers: nodes and node liveness statuses.
Reducers and sagas for these slices are implemented
in the same way as statements redux functionality,
with only one exception:
- nodes saga performs transformation for received
data to replicate the same logic as were before.

koorosh committed:
cluster-ui: fix modules path resolution when import from src/index.ts

"src/index.ts" is an entry point for bundled package and contains
exports all components. Importing modules inside this package
allows simplify import path, buy it causes another problem -
circular dependencies during module path resolution.

To avoid this problem, all import paths have been changed to point
directly to module.

koorosh committed:
cluster-ui: provide ordered list of diagnostics to statements page

Due to missed type definition for returned type for
`selectDiagnosticsReportsPerStatement` selector, it caused
a regression issue when instead of returning the list of all
diagnostics reports per statement, this selector returned only last report
(according to previous and outdated requirements).

Now, return type is explicitly defined for `selectDiagnosticsReportsPerStatement`
 selector and it returns aggregated statement stats with all diagnostics reports
 per statement.

We need all reports on Statements page to show dropdown list with current state
for recently requested diagnostics and links for previously generated reports.

koorosh committed:
cluster-ui: fix ts config for storybooks (#180)

Excluded .stories.tsx files from `tsconfig` caused issues that
ts rules weren't applied to storybooks and caused errors during
development (due to default ts rules were applied instead of
project specific rules from tsconfig.json file.
Exclusion of `.stories.tsx` files was intended to fix typings
generation for production build.

Final build has to include only sources from `./src` dir (this
dir is defined as a root dir in `tsconfig`), but
storybooks referenced to ./.storybook dir (which is outside of
`src` dir) and caused incorrect directory structure in `dist` dir.

With this change, all storybook custom decorators which is imported
within source code - moved inside `src` directory. This
structural change, provides proper output directory structure and
apply ts rules for all files in source directory.

As an additional change, storybook's backgroundColor decorator
is removed and instead, storybook configuration is provided with
default background color.

koorosh committed:
v0.1.41

Commit:
Publish (#183)

- @cockroachlabs/cluster-ui@0.1.42
 - @cockroachlabs/design-tokens@0.3.0
 - storybook-ui-components@0.2.14
 - @cockroachlabs/ui-components@0.2.16

koorosh committed:
v0.1.43

koorosh committed:
Revert "cluster-ui: nodes redux functionality"

This reverts commit 1fba624507dcce2892ece8b7da80c123dd7b2111.

koorosh committed:
v0.1.44

koorosh committed:
cluster-ui: aggregate node stats in selectors

Initially, received nodes response from server was modified
by accumulating node stats and then saved in redux store.

With this change, redux store keeps original response data, and
stats accumulation is executed in selectors when data requested.
The benefit of this is to have both original data on a client,
and cached transformed data as well.

koorosh committed:
v0.1.45

Commit:
# cluster-ui // Truncating long queries with CSS (#179)

* CRDB-2271: Truncating long queries with CSS

Applying a tried and true hack using `-webkit-line-clamp` to truncate
queries in the statements table to a couple of lines followed by an
ellipsis. To allow a longer preview, I am applying the same fix to the
tooltip, but clamping the lines to 15 (there were around 30 characters
per line and design asked for ~425 characters. It's close enough).

* Adding truncation to Transactions table

Since statements and transactions have enough variation to not be able
to make this change at the common table level, I'm adding truncation to
the transactions table styles to mirror the change in the statements
page table.

* Adding trucation to Statement col in Sessions

Adding trucation to the Statements column on the Sessions page to be
consistent with how statements appear on Statements and Tranactions
pages.

* Truncating long app names in Statements app filter

If a very long name is displayed in the Statements application filter,
the name may cause the dropdown to extend outside of the view or at
least beyond the table and cause excessive horizontal scrolling. To
correct this I have added the `nowrap` option to the `<Text>` component
inside of the Dropdown trigger. The "nowrap" prop will constrain the the
text to a single line but does not offer a constraint for the width. To
constrain the width I have added a local style set to a max-width of
`75ch` (75 characters wide).

Just a note, that I have used this opportunity to bring in the `<Text>`
from `ui-components` to replace the local one here, which was the
inspiration for the common component.

The width and display of the menu are controlled by the Dropdown
component and I didn't want to alter the display of all dropdowns that
use that component. So instead I extended the dropdown component to take
an additional `itemsClassname` prop to apply a class name to the element
rendered for the menu items. This way I can supply another local style
to constrain the width of the menu (to `80ch`) and apply text
truncation.

* Replacing repetitive styles with Sass mixins

DRYing up the code a little with two Sass mixins: text-overflow and
line-clamp.

Commit:
v0.1.46

dhartunian committed:
cluster-ui: fix lint errors in barCharts.tsx

Commit:
cluster-ui: update statements page

This commit adds columns for execution stats that we want to display for 21.1.
Namely, this adds rows/bytes read, max memory, and network bytes sent. The
rows affected column has been removed.
The commit also reorders all column-related code to follow the graphical order.

dhartunian committed:
cluster-ui: small cleanup of stmts table css

Make bar chart wrapper apply a simple set of
uniform styles instead of mandating a long
list of carefully selected selectors.

Adjust existing width overrides of certain
columns to not refer to the
"name"+`--bar-chart` selector and instead
target the bar charts inside of the specific
table cells. This reduces the need for
named bar chart selectors inside each column.

Commit:
v0.1.47

koorosh committed:
v0.1.48

koorosh committed:
cluster-ui: fix link on statement diagnostics bundle

Before, cluster-ui base path was defined as '/' root path and
links relied on it. But now, base pack can be set to any other
path and all requests on server should append updated base path.
`getBasePath` function returns current base path or empty string
if it is not set.

This change updates the link to use `getBasePath` and build
valid URL which includes base path.

koorosh committed:
v0.1.49

Commit:
cluster-ui: add Required to addStatementStats

Previously, the resullt of addStatementStats would have some fields unset. This
commit sets those fields and adds Required<StatementStatistics> so that any new
fields have to be handled by addStatementStats.

Commit:
cluster-ui: fix usage of wrong count in barCharts stddev calculation

stddev for rows and bytes read used the sample count, when the calculations
should use the statement count, since these stats are not sampled.

Commit:
cluster-ui: update transactions page with execution stats

This commit removes the rows affected column from the transactions page and
adds rows/bytes read, max memory usage, and latency. A contention column is
left for a future commit (pending backend changes).

Commit:
v0.1.50

koorosh committed:
cluster-ui: add babel-runtime to support dynamic loading

Before, when components from `cluster-ui` packages were dynamically imported
into client code, it might fail to load with an error "regeneratorRuntime is
not defined" error that explained in following `babel` issue
https://github.com/babel/babel/issues/8829.
As a solution, `babel-runtime` and `@babel/plugin-transform-runtime` plugin are
 added in package.

dependabot[bot] committed:
Bump ini from 1.3.5 to 1.3.7 (#157)

Bumps [ini](https://github.com/isaacs/ini) from 1.3.5 to 1.3.7.
- [Release notes](https://github.com/isaacs/ini/releases)
- [Commits](https://github.com/isaacs/ini/compare/v1.3.5...v1.3.7)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
v0.1.51

koorosh committed:
cluster-ui: bump cluster-ui to 0.2.x version

Latest changes to cluster-ui introduced incompatibility
between CRDB version 20.2 and ongoing development for 21.1.

Changes related to CockroachDb 21.x versions have to be
added to 0.2.x version and 0.1.x version will be targeted
to CockroachDb 20.2 version (for hotfixes, etc)

cluster-ui@0.1.x ver can be published from `cluster-ui-20.2`
branch.

koorosh committed:
v0.2.1

koorosh committed:
v0.2.2

vladlos committed:
cluster-ui: sessions redux functionality

minor cleanups + recreated redux layer for sessions page,
sessions details page, terminate query modal.

notification message support after success\error on terminate
query modal flow will be done as separate pr, due to very
different approach of working with notifications in `dbconsole`
and `managed-service`.

barryhe2000 committed:
cluster-ui: add full table or index scan filter to statements page

Add checkbox option to filter statements by full table or index scan.

vladlos committed:
v0.2.3

Commit:
cluster-ui: update transaction overview page with execution stats

dhartunian committed:
cluster-ui: update README to summarize dev workflow (#199)

Also added some diagrams to describe how the different components here
fit together.

dhartunian committed:
v0.2.4

Commit:
v0.2.5

Commit:
cluster-ui: add execution stats to statement details page

Commit:
v0.2.6

dhartunian committed:
v0.2.7

vladlos committed:
cluster-ui: fix appstat utils

additional null check to fix crashes in dbconsole envoirement

Commit:
cluster-ui: update crdb-protobuf-client

Commit:
v0.2.8

koorosh committed:
cluster-ui: clean up unused CSS import

cluster-ui had `design-tokens` import which
wasn't installed as dependency and caused failure
 during publishing.

koorosh committed:
v0.2.9

vladlos committed:
v0.2.10

Commit:
cluster-ui: remove TXN Type column in statements page

This column is no longer necessary because the transactions page exists.

Commit:
cluster-ui: fix sort order in statement/transaction table views

The default sort order is specified using a column index. Previous commits that
rearranged these columns did not update the index, so an erroneous sort order
was observed.

Commit:
cluster-ui: update "Latency" column titles to "{Statement,Transaction} Time"

Commit:
cluster-ui: use formatTwoPlaces when displaying row stat in transaction details

This limits the number of decimal places to two, achieving better aesthetic
consistency.

Commit:
cluster-ui: add max scratch disk usage to statement/transaction details

Commit:
cluster-ui: add contention column to statements/transactions pages

This commit adds a contention time column that displays the amount of time
a statement spent contending and the cumulative time a transaction's
statements spent contending cumulatively in each respective page.

Commit:
v0.2.11

koorosh committed:
cluster-ui: reset redux state action

Initially, it wasn't possible to reset redux state at some point
during application life cycle.
Switching between multiple clusters that should keep their own
state in redux store requires to have ability to re-initialize
an empty redux state.

barryhe2000 committed:
v0.2.12

vladlos committed:
cluster-ui: aggregate stmn data on txn details page

before: multiple copies of same statements was show on
transaction details page. copies have same id but different
node id.
after: aggregate function used, duplicates eliminated.added
basic unit test for aggregate method.

koorosh committed:
cluster-ui: fix styles for execution stats tables

Execution stats tab on statement details page contains several tables
that due to some regressions had broken styles and missed border colors,
header and footer backgrounds.

Current change recovers some lost styles and fixes stylus nested style rules.

Commit:
cluster-ui: display "no samples" and a tooltip when no stats have been sampled

This commit adds an option to the bar chart to display a message of
"no samples" accompanied with a tooltip when a sampled execution stat has not
been sampled yet. This is more user friendly as we would previously show a zero
value.

Commit:
cluster-ui: update tooltips on statements/transaction pages

koorosh committed:
v0.2.13

vladlos committed:
cluster-ui: sessions, terminate modal notifications

provide separate notifications saga and expose its action, so consumer
app can build around it its own notification handlers. small cleanups.

vladlos committed:
v0.2.14

Commit:
v0.2.15

vladlos committed:
v0.2.16

koorosh committed:
cluster-ui: fix styles for statements page

The problem appears when global styles for `span` and `a` elements can
override inherited styles in statements table.
To avoid this, styles for some of the elements are set explicitly to
avoid global styles overrides

koorosh committed:
cluster-ui: ui customization parameters

Before, it wasn't necessary to customize UI components depending
on specific runtime conditions.
But now, it has to be possible to allow some client application
code to define which components have to be hidden on UI or shown.

In this particular change, "Statement diagnostics" link has to be
hidden because an entire Statement Diagnostics page might not be
available in client's code.

UI customization is implemented within redux store as separate
slice where all required customization params can be defined
with their default values.
To change some params, client's application has to dispatch
uiConfig's "update" action with partially filled configuration.

koorosh committed:
cluster-ui: fix statements page styles

Different pages use different implementations of table component
with their very own styles or override default styles with custom.
It caused styles mismatching.

Current change doesn't try to unify all components/styles but
locally fixes styles and somehow reuses it. It'll allow to
backport changes for older versions with less stress.

maryliag committed:
add pct of all runtime to statements

koorosh committed:
v0.2.17

koorosh committed:
cluster-ui: correct design for empty states in statements table

Statements table and statement diagnostics table display empty state
as embedded placeholder inside of table (when table header is visible).
According to new design, empty state placeholders should replace entire
table.
Current change changes placeholders in statements table and diagnostics
tables. Also some refactoring is done in Empty panel component to reuse
components from `ui-components` package and clean up hand made styles.

koorosh committed:
v0.2.18

koorosh committed:
v0.2.19

koorosh committed:
cluster-ui: re-export uiConfiguration actions

cluster-ui package has to re-export uiConfig actions to allow external apps
to change the reducers state in type-safe way - with action creator and
provided types.

koorosh committed:
v0.2.20

maryliag committed:
v0.2.21

koorosh committed:
cluster-ui: proper empty state design for statement diagnostics view

Several commits ago, empty state component for Statement diagnostics view has been
changed to old design and now it is rolled back.

Additional changes were made to reuse `ui-components` (Heading, Text and Button).

koorosh committed:
v0.2.22

koorosh committed:
cluster-ui: default values for statement details ui configuration

Recently Statement details page were extended with `uiConfig` prop that
includes flags to tweak visibility for particular UI elements.
`uiConfig` prop should not be required, and it should be okay to
skip it to render default UI.

Current change adds default value for `uiConfig` prop and makes this
field as optional for statementDetailsProps type.

Release note: None

koorosh committed:
v0.2.23

maryliag committed:
cluster-ui: create component for filter and table stats and  use them on transaction and statements page

koorosh committed:
v0.2.24

koorosh committed:
cluster-ui: analytics tracking events

Current change extends list of tracked events and refactors
analytics reducer to make it more generic.

Analytics reducer provides single action (`track`) that accepts
different event types - it allows extend this action to work with
new types and have single to watch in case client's app is interested
in all possible analytics events.

Event types include required `name` field that has to follow Segment's
naming conventions and the rest of optional fields can be considered as
Segments properties.

To avoid coupling of analytics events with components - all logic are
defined in Connected components. Connected components have been refactored
`mapDispatchToProps` object is defined as function to be able to dispatch
multiple events in single method (`onActivateStatementDiagnostics` method
dispatches one action to create report and next action to track analytics).

Following events tracked:
- activate statements diagnostics (from statements and statement details pages)
- download statement diagnostics archive
- click on statement row in statements table
- change App filter on Statements page
- search statement on statements page
- change tabs on statement details page
- soring by columns in tables (statements table, and statement details
tables: Diagnostics and Stats table).
- track when click `Back to statements` button on statement details page;

maryliag committed:
v0.2.25

koorosh committed:
cluster-ui: fix table styles

Current change adjusts table styles according to latest
designs (that include headers with white backgrounds and
increased row height).

It affects following tables:
- Cluster overview page > Nodes table
- Sessions table
- Statements table
- Statement details page > Diagnostics table
- Statement details page > Execution stats table

In addition to table style fixes, following small changes were made:
- `Source Sans Pro` font is added as a required font from `ui` package
- used default link color for active tab on statement details page

koorosh committed:
v0.2.26

Azhng committed:
cluster-ui: add button to reset SQL stats on statement page and txn page

koorosh committed:
cluster-ui: cancel active sagas when redux state is reset

Previous implementation of redux-sagas for fetching new data from server was
straightforward and didn't take into account cases such as resetting an entire
redux state, so it cause cases when request to some data is dispatched, then
redux state is reset, and after this sagas would handle returned data as
usual (which in fact causes inconsistent store state).

To avoid this cases, sagas have to be cancelled when redux state is reset
and all responses after this should be ignored. To achieve desired behavior,
most of the changes are done within custom `throttleWithReset` effect.
- now, it's behavior is changed to take and process only latest actions
instead of processing every incoming action. It guaranties that all previous
actions and forked tasks are cancelled.
- `throttleWithReset` effect doesn't buffer actions and ignores all except
current action.

Sagas, that use `throttleWithReset` effect provide one more cancellation action
pattern to watch: `rootActions.resetState`. This action is dispatched when redux
state is reset and initiates cancellation process for running saga's tasks.

To follow the same pattern, statement diagnostics sagas are refactored in the
same was as other sagas, to use `throttleWithReset` effect.

koorosh committed:
v0.2.27

Azhng committed:
v0.2.28

Azhng committed:
v0.2.28

maryliag committed:
cluster-ui: last execution time for statement

maryliag committed:
v0.2.29

maryliag committed:
v0.2.30

maryliag committed:
cluster-ui: show database info and ability to choose statement columns

Information about database is now displayed on Statements page and
Statements Details page.
By default the column database is not displayed, but with the new column
selector option, it can be selected.
New filter allows to select statements based on database name.

Closes: #33316

Release note (ui change): Display database name information on Statements
page (hidden by default) and Statements Details page.

New filter option for statements based on databases name.

New option to select which columns to display on statements table.

Release note (bug fix): Transaction page showing correct value for implicit txn.

vladlos committed:
cluster-ui: clean up tooltip components (#251)

before: in `cluster-ui` package we used legacy versions of
tootips based on antd and with a bunch of code duplicates and
hacks.

after: used `ui-components` tooltip component, removed unneeded
code and styles. also on process of work find out that barchats
legend tooltips missing styles - added them.

use tooltip component from `ui-components` library instead local
versions based on antd. to achieve this `ui-componets/Tooltip`
also needed some updates:
 - provide new style for table title usage (it have same color
   scheme as `default` but different typographic style)
 - add ability to hover on tooltip for clickable elements inside
 - other minor style updates and fixes

as side effect of this work there is a fix of dev process,
updates of storybook and webpack config was needed to run build
of `cluster-ui` with linked local `ui-components`.

vladlos committed:
Revert "cluster-ui: clean up tooltip components (#251)" (#286)

This reverts commit 52f5bfaa7c0e076b7986889b8a124ef78cf13eb8.

vladlos committed:
cluster-ui: sessions style fixes (#275)

updated css to fix corresponding style issues that was caused by
global css of managed service. disabled link for gateway node on
sessions details page.

vladlos committed:
v0.2.31

vladlos committed:
cluster-ui: sessions tracking events (#278)

expose analytics events on sessions:
 - on table sorted
 - on session link clicked
 - on terminate session
 - on terminate statement
and sessions details page:
 - on back to sessions clicked
 - on terminate session
 - on terminate statement

vladlos committed:
v0.2.32

vladlos committed:
ui-components: tooltip component updates (#290)

provide new style for table title usage (it have same color
scheme as default but different typographic style)
add ability to hover on tooltip for clickable elements inside
other minor style updates and fixes
as side effect of this work there is a fix of dev process,
updates of storybook and webpack config was needed to run build
of cluster-ui with linked local ui-components

vladlos committed:
cluster-ui: sessions tracking (#288)

fix sessions table sort event payload

vladlos committed:
v0.2.33

vladlos committed:
cluster-ui: tooltip fixes (#292)

before: in cluster-ui package we used legacy versions of
tootips based on antd and with a bunch of code duplicates and
hacks.

after: used ui-components tooltip component, removed unneeded
code and styles. also on process of work find out that barchats
legend tooltips missing styles - added them.

vladlos committed:
v0.2.34

maryliag committed:
ui: fix typo on network tooltip

Fixing typo on the Network tooltip on the Statements Page.

Release note (ui change): Fix typo on Network tooltip on Statements Page.

maryliag committed:
v0.2.35

maryliag committed:
cluster-ui: adding information about node and region on statement details

Release note (ui change): Display on Statement Details information about
nodes and regions a statement was executed on.

maryliag committed:
cluster-ui: adding information about node and region on statement and transaction

Display information about which Regions/Nodes a statement was executed and
a filter option for region and another for nodes.

Release note (ui change): Display on Statement and Transaction pages information
about nodes and regions a statement was executed on.

(cherry picked from commit 088546c9d70fd44c75837af0801b3ea028ec4b95)

vladlos committed:
cluster-ui: bump ui-components (#299)

update ui-components dep that contain latest tooltip fixes

vladlos committed:
v0.2.36

maryliag committed:
v0.2.37

maryliag committed:
cluster-ui: return information about node regions to statement pages

Release note: None

maryliag committed:
v0.2.38

maryliag committed:
cluster-ui: returning information from cache about which region each node is on.

Part of issue cockroachdb/cockroach#64779

This will be followed by another PR with the changes on the ui visible to the user.

Release note: None

maryliag committed:
v0.2.39

maryliag committed:
v0.2.40

koorosh committed:
ui: integrate cluster-ui package into db-console

This change is a part of changes described in RFC for migrating cluster-ui into DB Console
 (https://github.com/cockroachdb/cockroach/pull/64085)

Initially db-console referenced to `cluster-ui` package as an external package that
is hosted in another repository and required to update version of this package
everytime new changes are made.

Now, when `cluster-ui` package is extracted from `cockroachdb/ui` repository and
moved in this repo (under `pkg/ui/cluster-ui`), it is possible to reference to
this package as a linked dependency and immediately observe all changes.

To do this following changes were made:
- db-console references to cluster-ui as a linked dependency and doesn't rely
on specific version
- `cluster-ui` package relies on local version of crdb-protobufjs-client package
instead of specific version
- Make commands related to db-console are extended to take into account nested
`cluster-ui` package. It runs tests, linting and watch commands for both packages.

One notable workaround has to mentioned... yarn installation for db-console
fails during first run because it cannot properly consume symlinked `cluster-ui`
dependency and only with second attempt it passes without issues (this is known
issue in yarn project).

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65126)
<!-- Reviewable:end -->

nkodali reviewed (approved):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>  Can you add a reference to the RFC in your commit message? Also adding @rickystewart as reviewer to look over Makefile changes.

> Reviewed 12 of 408 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65126#-:-M_fFZwyAsicZfDAqAda:b-yqnah9)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @elkmaster and @nathanstilwell)



<!-- Sent from Reviewable.io -->

rickystewart reviewed (approved):
`Makefile` looks fine :)

maryliag committed:
cluster-ui: fix sort on tables

Sort on tables of Statement, Transaction and Sessions page were
not using the right value from the url regarding sorting. It was
using the right sortKey, but always doing ascending. Fix done so
it uses the right value for the boolean.

Since the sortKet might be different for different users on Statement
Page (difference on columns being displayed), use the name of the
columns, instead of the key.

Fixes: https://github.com/cockroachdb/cockroach/issues/65085

Release note (bug fix): Sort on tables are now picking up the
correct value from the url.

(cherry picked from commit ab32cd46fe5db70147fe6eb1785d0d61ad294bdf)

maryliag committed:
v0.2.41

(cherry picked from commit a04919fcc3b95cb89fc51c553ae3650f0c32e2c0)

koorosh committed:
ui: add copyright headers to `cluster-ui` files

Files that were moved as part of cluster-ui package from `ui` repository
didn't have copyright headers that are required in current repo to
comply with lint rules.

This change appends copyright headers to all files in the package.

Release note: none

koorosh committed:
ui: fix typos in integrated ui/cluster-ui package

ui-lint command failed on misspelled words in the cluster-ui package
that was moved from `ui` repository.
Current change fixes typos to pass linting checks.

Release note: none

maryliag committed:
v0.2.42

(cherry picked from commit ca3ec8c2a59b996983c1bfe206a9e242edcc0d6a)

koorosh commented on pkg/ui/webpack.app.js:
```diff
@@ -30,21 +31,12 @@ class RemoveBrokenDependenciesPlugin {
   }
 }
 
-let DashboardPlugin;
```
@nathanstilwell , with current change, `make ui-watch` target will start two parallel webpack devservers (for cluster-ui and db-console) and I suggested that usage of Dashboard plugin isn't convenient here because it occupies entire screen and doesn't provide space for outputs from different processes.
Instead of it, I added `WebpackBar` plugin (the same as used in `cluster-ui`) to distinguish outputs.
What do you think about this change?

koorosh commented on Makefile:
```diff
@@ -1322,21 +1333,23 @@ bin/.gw_protobuf_sources: $(PROTOC) $(GW_SERVER_PROTOS) $(GW_TS_PROTOS) $(GO_PRO
 
-.SECONDARY: $(UI_TS_OSS) $(UI_TS_CCL)
-protos%.d.ts: protos%.js pkg/ui/yarn.installed
+.SECONDARY: $(UI_TS_CCL) $(UI_TS_OSS)
+$(UI_TS_CCL): $(UI_JS_CCL) pkg/ui/yarn.protobuf.installed
+$(UI_TS_OSS): $(UI_JS_OSS) pkg/ui/yarn.protobuf.installed
+$(UI_TS_CCL) $(UI_TS_OSS):
```
@rickystewart , I had to change the way targets are defined to add more specificity. I observed the problem when both CCL and OSS targets have to be built concurrently and as result it could use `protosccl.js` for both CCL and OSS targets.

Commit:
21.2 Version alignment (#319)

* cluster-ui: Increasing minor version to align with 21.2 development

Increasing the minor version of cluster-ui to represent the next
unreleased version of CRDB (21.2) that is being developed on the master
branch of `cockroachdb/cockroach`.

* cluster-ui: Adding Version explanation to README

There was no mention of how minor versions of cluster-ui are aligned to
major release versions of CRDB, so adding a section outlining this
mapping. Also documenting the versioning skew and deprecation that
happened after the CRDB 21.1 release.

(cherry picked from commit 20804e3ba9a877b080fb29c149794e824750ab80)

Commit:
v0.3.1

(cherry picked from commit 8303a77bb5a4bf0c5b28f05c5b9a5a339538333a)

koorosh commented:
bors r+

koorosh commented:
bors r+

koorosh commented:
bors r+

koorosh commented:
bors r+
